numpy/_core/src/multiarray/item_selection.c

```diff
@@ -2525,11 +2526,13 @@ count_nonzero_u8(const char *data, npy_intp bstride, npy_uintp len)
         len  -= len_m;
         count = len_m - zcount;
     #else
-        if (!NPY_ALIGNMENT_REQUIRED || npy_is_aligned(data, sizeof(npy_uint64))) {
```

Comment:
You still need `npy_is_aligned(data, sizeof(npy_uint64))`

Reply:
The idea is to get rid of all the pointer aliasing logic so we can finally be strict-aliasing valid